analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.cs

```diff
@@ -63,16 +67,55 @@ public void SequentialInput_VB()
         {
             var context = SETestContext.CreateVB("Dim A As Boolean = True, B As Boolean = False : B = Not B : A = (B)");
             context.Collector.ValidateOrder(
+                "LocalReference: A (Implicit)",
                 "Literal: True",
+                "SimpleAssignment: A As Boolean = True (Implicit)",
+                "LocalReference: B (Implicit)",
                 "Literal: False",
+                "SimpleAssignment: B As Boolean = False (Implicit)",
                 "LocalReference: B",
                 "LocalReference: B",
                 "UnaryOperator: Not B",
+                "SimpleAssignment: B = Not B (Implicit)",
                 "ExpressionStatement: B = Not B",
                 "LocalReference: A",
                 "LocalReference: B",
                 "Parenthesized: (B)",
+                "SimpleAssignment: A = (B) (Implicit)",
                 "ExpressionStatement: A = (B)");
         }
+
+        [TestMethod]
+        public void PreProcess_Null_StopsExecution()
```

Comment:
open, possibly silly question: is there a way to design the system without relying on null?

Reply:
It is, but `null` is the best and simplest option for now. Other options:
* exception driven (baaad)
* array-based (not needed, see https://github.com/SonarSource/sonar-dotnet/pull/5100#discussion_r758575062)
* flag-based (but `flag` would be redundant to `state==null` anyway => confusing redundancy)

Reply:
I was suggesting more an alternative State which no Empty but maybe `None` . 

Because `null` can lead to confusion too, as it can either be in  the program at some point on purpose or as a non-initialized variable.

Also, a program state None can also contain some additional meta-information which can be used to debug (e.g. what was the context in which it was initially returned)

Reply:
I prefer the `null` variant. We don't need to deal with the question `what is the difference between null and None?`

Reply:
well, if we design a system in which null should not exist:
- `None` is intended
- `null` is a bug